experimental/dds/tree2/src/feature-libraries/sequence-field/moveEffectTable.ts

```diff
@@ -97,6 +97,14 @@ export function isMoveIn(effect: MarkEffect): effect is MoveIn {
 	return effect.type === "MoveIn";
 }
 
+export function getMoveIn(effect: MarkEffect): MoveIn | undefined {
+	return effect.type === "MoveIn"
+		? effect
+		: effect.type === "AttachAndDetach"
+		? getMoveIn(effect.attach)
+		: undefined;
```

Comment:
I find this somewhat hard to read. I would prefer if the inner ternary cases were idented, but it seems that Prettier doesn't like that.

Reply:
A switch on `effect.type` might be cleaner.